canvas_modules/common-canvas/src/common-canvas/svg-canvas-renderer.js

```diff
@@ -3330,48 +3376,21 @@ export default class SVGCanvasRenderer {
 
 	// Sets the image passed in into the D3 image selection passed in. This loads
 	// svg files as inline SVG while other images files are loaded with href.
-	// We will only set a new image if the image is new or has changed from what
-	// it was previously set to. This allows applications to set new images while
-	// the canvas is being displayed.
 	setImageContent(imageSel, image) {
-		const nodeImageType = this.getImageType(image);
 		if (image !== imageSel.attr("data-image")) {
+			const nodeImageType = this.getImageType(image);
 			// Save image field in DOM object to avoid unnecessary image refreshes.
 			imageSel.attr("data-image", image);
 			if (nodeImageType === "svg") {
-				this.addCopyOfImageToDefs(image);
-				this.updateUseObject(imageSel, image);
+				d3.xml(image, { cache: "force-cache" }).then((data) => {
```

Comment:
Are the images still stored in `defs`?

Reply:
No unfortunately with that approach we can only style icons once no matter how many times they are referenced from nodes. This means if we had two nodes both referring to the same icon we couldn't style one with fill = "grey" and one that is not grey. So now I include the SVG in the DOM for each node so in the example above the SVG would be inserted twice; once for each node. This means the icons can be styles separately on  a node by node basis.